service/src/main/frontend/src/pages/DataDictionary.vue

```diff
@@ -0,0 +1,287 @@
+                )
+              }}
+              </label>
+            </q-td>
+          </q-tr>
+        </template>
+      </q-table>
+    </div>
+  </main>
+  <div v-if="banner?.enabled" :style="banner?.styleTop" style="margin-top: 0.50vh;">
+      {{ banner?.messageTop }}
+  </div>
+  <div v-if="banner?.enabled" :style="banner?.styleBottom">
+      {{ banner?.messageBottom }}
+  </div>
+</template>
+
+<script setup lang="ts">
+import { onMounted, ref } from 'vue';
+import { QTable, QTableProps, exportFile, useQuasar, Notify } from 'quasar';
+import { useToggle, useDark } from '@vueuse/core';
+import { api } from '../boot/axios';
+import { Banner, columns } from '../functions/components';
+import * as Formatters from '../functions/formatters';
+import * as Wrapper from '../functions/csvWrapper';
+import * as Feature from '../functions/features';
+
+// Defines the Table References, loading for axios, search filter, and pagination to sort.
+const $q = useQuasar();
+const table = ref();
+const loading = ref(true);
+const filter = ref('');
+const changeFilter = ref('');
+const banner = ref<Banner>();
+let rows: QTableProps['rows'] = [];
+let paginationFront = ref({
+  rowsPerPage: 200,
+  sortBy: 'fieldName',
+});
+
+// API - Defines all the Nececssary API calls for the user, and filters.
+onMounted(() => {
+  api
+  .get('banner/', undefined)
+  .then((response) => {
+    banner.value = response.data as Banner;
+  })
+  .catch((reason) => {
+    console.log('Could not fetch banner: ' + reason);
+  });
+
+  api
+  .get('')
+  .then((response) => {
+    // Mini Filter to sort collapsable Rows
+    rows = response.data.MetadataFields.sort((a: any, b: any) => {
+      // Create a combined key for both fieldname and internalFieldName
+      const keyA = `${a.fieldname}_${a.internalFieldName}`;
+      const keyB = `${b.fieldname}_${b.internalFieldName}`;
+
+      // First compare the combined key alphabetically
+      if (keyA < keyB) {
+        return -1;
+      } else if (keyA > keyB) {
+        return 1;
+      } else {
+        // If the combined key is the same, compare lastUpdated in descending order
+        return b.lastUpdated - a.lastUpdated;
+      }
+    });
+    rows = Formatters.setVisibility(rows);
+    loading.value = false;
+  })
+  .catch((reason) => {
+    console.log('Error fetching and formatting rows: ' + reason);
+  });
+});
+
+// Export - Attempts to Wrap the CSV and Download.
+function exportTable(this: any) {
+  const rowsToExport = table.value?.filteredSortedRows.filter(
+    Formatters.isVisible
+  );
+  const content = [columns!.map((col) => Wrapper.wrapCsvValue(col.label))]
+    .concat(
+      rowsToExport.map((row: any) =>
+        columns!
+          .map((col: any) =>
+            Wrapper.wrapCsvValue(
+              typeof col.field === 'function'
+                ? col.field(row)
+                : row[col.field === void 0 ? col.name : col.field],
```

Comment:
Yeah its working: `void 0` in Javascript essentially means `undefined` in Typescript. I'll swap it to use `undefined` since majority of the code is Typescript.